### PR TITLE
[Cleanup] Moved test helper code from "test/support" to a new "devtools_test" package

### PR DESCRIPTION
**Pull request by matt-ragonese**
In addition to creating the new `devtools_test` package and moving `test/support` code into it, this change also:
- moves shared mocks/fakes in `devtools_app` into `devtools_test`
- moves the `waitFor(...)` helper function into `devtools_test`'s `utils.dart`
- adds a new `devtools_app.dart` file that exports `src` files used outside of `devtools_app`

### CODE REVIEW

File: packages/devtools_test/.flutter-plugins-dependencies [hunk 1]
```diff
@@ -0,0 +1 @@
...s-0.0.2+1/","dependencies":[]},{"name":"url_launcher_windows","path":"/Users/mragonese/Documents/ExternalSDKs/flutter/.pub-cache/hosted/pub.dartlang.org/url_launcher_windows-0.0.1+3/","dependencies":[]}],"web":[{"name":"file_selector_web","path":"/Users/mragonese/Documents/ExternalSDKs/flutter/.pub-cache/hosted/pub.dartlang.org/file_selector_web-0.8.1+2/","dependencies":[]},{"name":"url_launcher_web","path":"/Users/mragonese/Documents/ExternalSDKs/flutter/.pub-cache/hosted/pub.dartlang.org/url_launcher_web-0.1.5+3/","dependencies":[]}]},"dependencyGraph":[{"name":"file_selector","dependencies":["file_selector_web"]},{"name":"file_selector_linux","dependencies":[]},{"name":"file_selector_macos","dependencies":[]},{"name":"file_selector_web","dependencies":[]},{"name":"file_selector_windows","dependencies":[]},{"name":"url_launcher","dependencies":["url_launcher_web","url_launcher_linux","url_launcher_macos","url_launcher_windows"]},{"name":"url_launcher_linux","dependencies":[]},{"name":"url_launcher_macos","dependencies":[]},{"name":"url_launcher_web","dependencies":[]},{"name":"url_launcher_windows","dependencies":[]}],"date_created":"2021-09-28 14:01:53.696333","version":"2.5.1"}
```
Comment (kenzieschmoll) [inline packages/devtools_test/.flutter-plugins-dependencies:1]: this file should not be checked in. There is an example of adding this to the .gitignore file under devtools_app/.gitignore

File: packages/devtools_test/CHANGELOG.md [hunk 2]
```diff
@@ -0,0 +1,2 @@
+## 2.6.1-dev.1
```
Comment (kenzieschmoll) [inline packages/devtools_test/CHANGELOG.md:1]: version should be dev.2

File: packages/devtools_test/lib/utils.dart [hunk 3]
```diff
@@ -297,3 +294,23 @@ void verifyIsSearchMatch(
     }
   }
 }
+
+Future<void> waitFor(
```
Comment (kenzieschmoll) [inline packages/devtools_test/lib/utils.dart:298]: this looks like it needs to be formatted with dartfmt

File: packages/devtools_test/.flutter-plugins-dependencies (hunk 1 above)
Comment (matt-ragonese) [reply packages/devtools_test/.flutter-plugins-dependencies]: Whoops, thanks for catching! Removed.

File: packages/devtools_test/CHANGELOG.md (hunk 2 above)
Comment (matt-ragonese) [reply packages/devtools_test/CHANGELOG.md]: Thanks, fixed!

File: packages/devtools_test/pubspec.yaml [hunk 4]
```diff
@@ -0,0 +1,53 @@
+name: devtools_test
+description: A package containing shared test helpers for Dart DevTools tests.
+
+# Note: this version should only be updated by running tools/update_version.dart
```
Comment (kenzieschmoll) [inline packages/devtools_test/pubspec.yaml:4]: you'll need to add this pubspec path to the update_version.dart script

File: packages/devtools_test/lib/utils.dart (hunk 3 above)
Comment (matt-ragonese) [reply packages/devtools_test/lib/utils.dart:298]: Thanks, just updated this after running dartfmt from packages/ directory.

File: packages/devtools_test/pubspec.yaml (hunk 4 above)
Comment (matt-ragonese) [reply packages/devtools_test/pubspec.yaml:4]: Thanks! Updated `update_version.dart`.

File: packages/devtools_test/pubspec.yaml [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+  mockito: ^5.0.9
+  webkit_inspection_protocol: '>=0.5.0 <2.0.0'
+
+dependency_overrides:
+  # The '#OVERRIDE_FOR_DEVELOPMENT' lines are stripped out when we publish.
+  # All overriden dependencies are published together so there is no harm
+  # in treating them like they are part of a mono-repo while developing.
+  devtools_app: #OVERRIDE_FOR_DEVELOPMENT
+    path: ../devtools_app #OVERRIDE_FOR_DEVELOPMENT
+  devtools_server: #OVERRIDE_FOR_DEVELOPMENT
```
Comment (kenzieschmoll) [inline packages/devtools_test/pubspec.yaml:50]: we can remove this, as we don't depend on devtools_server above

Comment (matt-ragonese) [reply packages/devtools_test/pubspec.yaml]: Thanks! Removed this (as well as other unused deps in this file).

File: .gitignore [hunk 6]
```diff
@@ -20,4 +20,4 @@ flutter-sdk/
 .flutter-plugins
 flutter_export_environment.sh
 Podfile
-Podfile.lock
+Podfile.lock
```
Comment (kenzieschmoll) [inline .gitignore:23]: still seeing an accidental commit here

Comment (matt-ragonese) [reply .gitignore]: Removed.

**Review by kenzieschmoll — APPROVED** [review]
LGTM

**Pull request merged**